### ISSUE TITLE
control environment passed to auth exec

### ISSUE THREAD

**Issue by aviramha** [kubernetes/client-go#1177]
Right now in the `ExecConfig` struct you can add env vars using the `Envs` field but by default it merges with the host process environment variable, which might be not desirable (for example if the host process has `AWS_REGION=X` and that's not what the user wanted the k8s to have)
Our specific use case is described here https://github.com/kube-rs/kube/issues/1060
but maintainers there suggested we should align upstream.
Basically the suggestion is to add to `ExecConfig` a list of envs to remove before running the process, `drop_env` as a suggested name. User of the go client library can specify it to avoid leak of env.
